panic in the HistoricalStatsWorkerLoop

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)

### 3. What did you see instead (Required)

```
[2023/12/01 03:12:52.070 +08:00] [ERROR] [misc.go:116] ["panic in the recoverable goroutine"] [label=domain] [funcInfo=HistoricalStatsWorkerLoop] [r="runtime error: invalid memory address or nil pointer dereference"] [stack="github.com/pingcap/tidb/pkg/util.Recover\n\t/home/jenkins/agent/workspace/pingcap/tidb/ghpr_check2/tidb/pkg/util/misc.go:120\nruntime.gopanic\n\t/usr/local/go/src/runtime/panic.go:914\nruntime.panicmem\n\t/usr/local/go/src/runtime/panic.go:261\nruntime.sigpanic\n\t/usr/local/go/src/runtime/signal_unix.go:861\ngithub.com/pingcap/tidb/pkg/statistics/handle/history.RecordHistoricalStatsToStorage\n\t/home/jenkins/agent/workspace/pingcap/tidb/ghpr_check2/tidb/pkg/statistics/handle/history/history_stats.go:132\ngithub.com/pingcap/tidb/pkg/statistics/handle/history.(*statsHistoryImpl).RecordHistoricalStatsToStorage.func1\n\t/home/jenkins/agent/workspace/pingcap/tidb/ghpr_check2/tidb/pkg/statistics/handle/history/history_stats.go:59\ngithub.com/pingcap/tidb/pkg/statistics/handle/util.WrapTxn\n\t/home/jenkins/agent/workspace/pingcap/tidb/ghpr_check2/tidb/pkg/statistics/handle/util/util.go:185\ngithub.com/pingcap/tidb/pkg/statistics/handle/util.CallWithSCtx\n\t/home/jenkins/agent/workspace/pingcap/tidb/ghpr_check2/tidb/pkg/statistics/handle/util/util.go:101\ngithub.com/pingcap/tidb/pkg/statistics/handle/history.(*statsHistoryImpl).RecordHistoricalStatsToStorage\n\t/home/jenkins/agent/workspace/pingcap/tidb/ghpr_check2/tidb/pkg/statistics/handle/history/history_stats.go:58\ngithub.com/pingcap/tidb/pkg/domain.(*HistoricalStatsWorker).DumpHistoricalStats\n\t/home/jenkins/agent/workspace/pingcap/tidb/ghpr_check2/tidb/pkg/domain/historical_stats.go:81\ngithub.com/pingcap/tidb/pkg/domain.(*Domain).StartHistoricalStatsWorker.func1\n\t/home/jenkins/agent/workspace/pingcap/tidb/ghpr_check2/tidb/pkg/domain/domain.go:2145\ngithub.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run.func1\n\t/home/jenkins/agent/workspace/pingcap/tidb/ghpr_check2/tidb/pkg/util/wait_group_wrapper.go:99"]
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->

--- a/pkg/statistics/handle/history/history_stats.go
+++ b/pkg/statistics/handle/history/history_stats.go
@@ -53,7 +53,10 @@ func (sh *statsHistoryImpl) RecordHistoricalStatsToStorage(dbName string, tableI
 	if err != nil {
 		return 0, errors.Trace(err)
 	}
-
+	if js == nil {
+		logutil.BgLogger().Warn("no stats data to record", zap.String("dbName", dbName), zap.String("tableName", tableInfo.Name.O))
+		return 0, nil
+	}
 	var version uint64
 	err = util.CallWithSCtx(sh.statsHandle.SPool(), func(sctx sessionctx.Context) error {
 		version, err = RecordHistoricalStatsToStorage(sctx, physicalID, js)

